release-22.2: awsdms: retry testing the connection on StartReplicationTask

otan opened the pull request:
Backport 1/1 commits from #101621.

/cc @cockroachdb/release

---

DMS seems to sometimes not start up because it thinks the tested connection wasn't successful (despite it clearly being successful earlier). Counter this for now by retrying.

Release note: None
Informs: https://github.com/cockroachdb/cockroach/issues/101417, https://github.com/cockroachdb/cockroach/issues/101517, https://github.com/cockroachdb/cockroach/issues/100322
Release justification: test only fix

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101699)
<!-- Reviewable:end -->

otan committed:
awsdms: retry testing the connection on StartReplicationTask

DMS seems to sometimes not start up because it thinks the tested
connection wasn't successful (despite it clearly being successful
earlier). Counter this for now by retrying.

Release note: None
Informs: https://github.com/cockroachdb/cockroach/issues/101417
